### PR TITLE
Feature/single file structure

### PR DESCRIPTION
**Pull request by KaleabTessera**
## What?
- Added basic structure for single file implementations. 
- Added requirements and dockerfile.  
## Why?
-
## How?
-
## Extra
Closes https://github.com/instadeepai/Mava/issues/885 .

### CODE REVIEW

**Review by sash-a — APPROVED** [review]
Thanks for this @KaleabTessera, looks great! 

Just one thing I would change is the filename ddpg to iddpg, just so it's not confused with a single agent version and to keep it consistent with ippo and idqn, but we can do this in our implementation :smile:

**Comment by KaleabTessera** [comment]
> Thanks for this @KaleabTessera, looks great!
> 
> Just one thing I would change is the filename ddpg to iddpg, just so it's not confused with a single agent version and to keep it consistent with ippo and idqn, but we can do this in our implementation smile

Thanks for the review @sash-a ! I am happy for the people doing the implementation to rename it. I just haven't seen `iddpg` anywhere, so maybe when implementing, people can research the name.

**Review by siddarthsingh1 — CHANGES_REQUESTED** [review]
Thanks @KaleabTessera , looking forward to working on this. Just a quick thing though. I don't think we need both idqn and maqdn. They usually refer to the same algorithm in literature as far as I'm aware.

**Comment by KaleabTessera** [comment]
> Thanks @KaleabTessera , looking forward to working on this. Just a quick thing though. I don't think we need both idqn and maqdn. They usually refer to the same algorithm in literature as far as I'm aware.

So madqn, I was thinking DQN with joint action value function across all agents, but that is not quite common or part of ctde, so I changed it to rdqn (recurrent dqn).

**Review by siddarthsingh1 — DISMISSED** [review]
@KaleabTessera , thanks for the update.

File: educational/systems/random_agent.py [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+            random_seed=config.seed,
+        )
+    elif config.type == "smac":
+        env, _ = smac_utils.make_environment(
+            map_name=config.env_name, random_seed=config.seed
+        )
+    return env, config
+
+
+def make_system(
```
Comment (sash-a) [inline educational/systems/random_agent.py:127]: I still think this is too much boilerplate, but I'll approve here and we can deal with it in #887

**Review by sash-a — APPROVED** [review]
LGTM other than the random agent boilerplate, but would rather deal with that in another PR

File: educational/systems/random_agent.py (hunk 1 above)
Comment (KaleabTessera) [reply educational/systems/random_agent.py]: I accidentally added this here. I have removed it, it should just be in the random agent PR.

**Review by sash-a — APPROVED** [review]
Thanks for removing the random agent stuff, looks great!

**Review by siddarthsingh1 — APPROVED** [review]
Looks great! @KaleabTessera.

**Pull request merged**